src/combinatorics/set_family/intersecting.lean

```diff
@@ -0,0 +1,196 @@
+  { rintro rfl,
+    exact intersecting_singleton.1 h rfl },
+  { have := h.1.eq ha hb (not_not.2 hab),
+    rw [this, disjoint_self] at hab,
+    rw hab at hb,
+    exact h.2 (eq_singleton_iff_unique_mem.2
+      ⟨hb, λ c hc, not_ne_iff.1 $ λ H, h.1 hb hc H.symm disjoint_bot_left⟩) }
+end
+
+protected lemma subsingleton.intersecting (hs : s.subsingleton) : s.intersecting ↔ s ≠ {⊥} :=
+intersecting_iff_pairwise_not_disjoint.trans $ and_iff_right $ hs.pairwise _
+
+lemma intersecting_iff_eq_empty_of_subsingleton [subsingleton α] (s : set α) :
+  s.intersecting ↔ s = ∅ :=
+begin
+  refine subsingleton_of_subsingleton.intersecting.trans
+    ⟨not_imp_comm.2 $ λ h, subsingleton_of_subsingleton.eq_singleton_of_mem _, _⟩,
+  { obtain ⟨a, ha⟩ := ne_empty_iff_nonempty.1 h,
+    rwa subsingleton.elim ⊥ a },
+  { rintro rfl,
+    exact (set.singleton_nonempty _).ne_empty.symm }
+end
+
+/-- Maximal intersecting families are upper sets. -/
+protected lemma intersecting.is_upper_set (hs : s.intersecting)
+  (h : ∀ t : set α, t.intersecting → s ⊆ t → s = t) :
+  is_upper_set s :=
+begin
+  classical,
+  rintro a b hab ha,
+  rw h (insert b s) _ (subset_insert _ _),
+  { exact mem_insert _ _ },
+  exact hs.insert (mt (eq_bot_mono hab) $ hs.ne_bot ha)
+    (λ c hc hbc, hs ha hc $ hbc.mono_left hab),
+end
+
+/-- Maximal intersecting families are upper sets. Finset version. -/
+lemma intersecting.is_upper_set' {s : finset α} (hs : (s : set α).intersecting)
+  (h : ∀ t : finset α, (t : set α).intersecting → s ⊆ t → s = t) :
+  is_upper_set (s : set α) :=
+begin
+  classical,
+  rintro a b hab ha,
+  rw h (insert b s) _ (finset.subset_insert _ _),
+  { exact mem_insert_self _ _ },
+  rw coe_insert,
+  exact hs.insert (mt (eq_bot_mono hab) $ hs.ne_bot ha)
+    (λ c hc hbc, hs ha hc $ hbc.mono_left hab),
+end
+
+end semilattice_inf
+
+lemma intersecting.exists_mem_set {𝒜 : set (set α)} (h𝒜 : 𝒜.intersecting) {s t : set α}
+  (hs : s ∈ 𝒜) (ht : t ∈ 𝒜) : ∃ a, a ∈ s ∧ a ∈ t :=
+not_disjoint_iff.1 $ h𝒜 hs ht
+
+lemma intersecting.exists_mem_finset [decidable_eq α] {𝒜 : set (finset α)} (h𝒜 : 𝒜.intersecting)
+  {s t : finset α} (hs : s ∈ 𝒜) (ht : t ∈ 𝒜) : ∃ a, a ∈ s ∧ a ∈ t :=
+not_disjoint_iff.1 $ disjoint_coe.not.2 $ h𝒜 hs ht
+
+variables [boolean_algebra α]
+
+lemma intersecting.not_compl_mem {s : set α} (hs : s.intersecting) {a : α} (ha : a ∈ s) : aᶜ ∉ s :=
+λ h, hs ha h disjoint_compl_right
+
+lemma intersecting.not_mem {s : set α} (hs : s.intersecting) {a : α} (ha : aᶜ ∈ s) : a ∉ s :=
+λ h, hs ha h disjoint_compl_left
+
+variables [fintype α] {s : finset α}
+
+lemma intersecting.card_le (hs : (s : set α).intersecting) : 2 * s.card ≤ fintype.card α :=
+begin
+  classical,
+  refine (s ∪ s.map ⟨compl, compl_injective⟩).card_le_univ.trans_eq' _,
+  rw [two_mul, card_union_eq, card_map],
+  rintro x hx,
+  rw [finset.inf_eq_inter, finset.mem_inter, mem_map] at hx,
+  obtain ⟨x, hx', rfl⟩ := hx.2,
+  exact hs.not_compl_mem hx' hx.1,
+end
+
+variables [nontrivial α]
```

Comment:
Perhaps it's worth saying that the results below this fail if `α` has a unique element (and are uninteresting if it's empty) to justify this assumption